prober/prober_test.go

```diff
@@ -85,6 +86,30 @@ func TestProber(t *testing.T) {
 		assert.True(t, res.Probes[1].Duration > 190*time.Millisecond)
 		assert.False(t, res.Probes[1].Healthy)
 	})
+
+	t.Run("With a single probe that times out", func(t *testing.T) {
+		p := NewProber(WithTimeout(200 * time.Millisecond))
+		p.AddProbe(sampleprobe.NewTimedSampleProbe("test", true, 300*time.Millisecond))
+		start := time.Now()
+		res := p.CheckOneProbe(ctx, "test")
+		duration := time.Since(start)
+
+		assert.True(t, duration > 200*time.Millisecond)
+		assert.False(t, res.Healthy)
+		assert.Equal(t, res.Comment, "error")
+		assert.Equal(t, res.Error.Error(), "probe check failed: prober: context deadline exceeded")
```

Comment:
The expected value comes first (https://pkg.go.dev/github.com/stretchr/testify/assert#Equal):

```suggestion
		assert.Equal(t, "error", res.Comment)
		assert.Equal(t, "probe check failed: prober: context deadline exceeded", res.Error.Error())
```